src/utils/cellSetOperations.js

```diff
@@ -33,4 +33,32 @@ const intersection = (listOfSets, properties) => {
   return intersectionSet;
 };
 
-export { union, intersection };
+const complement = (listOfSets, properties) => {
+  if (!listOfSets) {
+    return new Set();
+  }
+
+  const selectedCells = listOfSets.map(
+    (key) => properties[key]?.cellIds || null,
+  ).filter(
+    (set) => set && set.size > 0,
+  ).reduce(
+    (acc, curr) => new Set([...acc, ...curr]),
+  );
+
+  const complementSet = Object.values(properties).map(
+    (cluster) => cluster.cellIds,
+  ).filter(
+    (set) => set && set.size > 0,
+  ).reduce(
+    (acc, curr) => new Set([...acc, ...curr]),
+  );
```

Comment:
This makes the assumption that all the cells will be assigned a cell set at some point during the analysis. This seems reasonable to me (we will have metadata tracks for sample, for example, which should be a label for all cells), but it is an assumption we are making here.

Can you add a comment to this function to mention this?

Reply:
Ok, I'll add the comments. By the way, aren't all cells included in at least 1 set?

Reply:
Comment on assumption added